api/pkg/validation/errors.go

```diff
@@ -0,0 +1,227 @@
+package validation
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func (v ErrorList) Find(field string) ErrorList {
+	errs := ErrorList{}
+	for _, err := range v {
+		if err.Field == field {
+			errs = append(errs, err)
+		}
+	}
+	return errs
+}
+
+func (v ErrorList) HasError(field string, errType ErrorType) bool {
+	for _, err := range v {
+		if err.Field == field && err.Type == errType {
+			return true
+		}
+	}
+	return false
+}
+
+// Error is an implementation of the 'error' interface, which represents a
+// field-level validation error.
+type Error struct {
+	Type     ErrorType
+	Field    string
+	BadValue interface{}
+	Detail   string
+}
+
+var _ error = &Error{}
+
+// Error implements the error interface.
+func (v *Error) Error() string {
+	return fmt.Sprintf("%s: %s", v.Field, v.ErrorBody())
+}
+
+// ErrorBody returns the error message without the field name.  This is useful
+// for building nice-looking higher-level error reporting.
+func (v *Error) ErrorBody() string {
+	var s string
```

Comment:
I think you can just do `fmt.Sprintf("%#v`, v)`

Reply:
Not sure what you mean here. Should I overwrite the entire method with that oneliner?
It's not my code btw, its from kubernetes